src/common/file_stream.cpp

```diff
@@ -45,8 +47,12 @@ InputFileStream::InputFileStream(const std::string &file)
 
   // insert .gz decompression
   if(marian::utils::endsWith(file, ".gz")) {
+#if defined(DECODER_ONLY)
+    ABORT(".gz file decompression not supported in decoder-only build of Marian: {}", file);
```

Comment:
That was my point, of sorts. DECODER_ONLY should not be synonymous with "This is a wasm compile". My suggestion would be to use cmake dependent options to set specific options that enable or disable certain features depending on the platform environment. If wasm compiles imply that zlib is not available, I'm fine with that, but

1. options should do what they say (e.g., NO_ZLIB)
2. they should be implemented via dependent options, so that we don't need a whole slew of -DWHATEVER={on|off} when running cmake.

So if the cmake determines that this is a wasm compile (e.g. by looking at the compiler executable name) and sets a slew of options (that do what they say), I'm happy with this. However, I can imagine a scenario (e.g. a translation server) where I don't want to compile and link a pile of executables that I won't need in the end, but U still want to be able to decompress .gz files on the fly.

Reply:
> options should do what they say (e.g., NO_ZLIB)
> they should be implemented via dependent options, so that we don't need a whole slew of -DWHATEVER={on|off} when running cmake.

@ugermann I agree with you and I will make changes 👍

Reply:
With an eye to consistency of naming of options used, I'd actually suggest to use USE_ZLIB rather than NO_ZLIB. It's easier to remember them that way.

Reply:
I changed it to WASM as .gz compression is unsupported only for wasm builds. 👍